knowlang/configs/base.py

```
@@ -1,18 +1,47 @@
-from pydantic_settings import SettingsConfigDict
+import os
+import sys
 from pathlib import Path
 from typing import Optional
+
 from pydantic import ValidationInfo
-import os
-import sys
+from pydantic_settings import SettingsConfigDict
+
+
+def get_resource_path(relative_path: str, default_path: Optional[str] = None) -> Path:
+    """Get absolute path to resource, works for dev and PyInstaller
+    
+    Args:
+        relative_path: The relative path to the resource
+        default_path: Optional alternative relative path to use if the primary path doesn't exist
+        
+    Returns:
+        Path: The absolute path to the resource (may not exist)
+    """
 
-def get_resource_path(relative_path: str) -> Path:
-    """Get absolute path to resource, works for dev and PyInstaller"""
     if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
         # Running in PyInstaller bundle
         base_path = Path(sys._MEIPASS)
     else:
         # Running in non-PyInstaller environment
         base_path = Path.cwd()
+
+    # Define paths to try: primary path first, then default path if provided
+    paths_to_try = [relative_path]
+    if default_path:
+        paths_to_try.append(default_path)
+
+    for path_to_try in paths_to_try:
+        full_path = base_path / path_to_try
+        
+        if full_path.exists():
+            return full_path
+    
+    # If we get here, none of the paths exist
+    # attempted_paths = [str(Path.cwd() / path) for path in paths_to_try]
+    # raise FileNotFoundError(
+    #     f"Resource not found. Attempted paths: {attempted_paths}"
+    # )
```

Sounds reasonable to raise error instead of returning non-existing path.